docs/Guide/preconditions/channel-types.mdx

```diff
@@ -3,4 +3,78 @@ sidebar_position: 3
 title: Setting the types of channel a command can run in
 ---
 
-## TODO
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+The [`runIn`][runin] command option can be used to specify the types of channels a command can run in. This can be
+useful if you're developing a command that, for example, displays the roles of a user. In that scenario, you'll want to
+make sure that the command can only be run in guild channels.
+
+:::info
+
+[Valid values][runintypes] for `runIn` are `'DM'`, `'GUILD_TEXT'`, `'GUILD_NEWS'`, `'GUILD_NEWS_THREAD'`,
+`'GUILD_PUBLIC_THREAD'`, `'GUILD_PRIVATE_THREAD'`, and `'GUILD_ANY'`
+
+:::
+
+<Tabs groupId="language-choice">
+<TabItem value="javascript" label="JavaScript" default>
+
+```javascript {7}
+const { Command } = require('@sapphire/framework');
+
+module.exports = class RolesCommand extends Command {
+  constructor(context) {
+    super(context, {
+      // ...
+      runIn: 'GUILD_ANY' // Only run in guild channels
+    });
+  }
+};
+```
+
+</TabItem>
+<TabItem value="ESM" label="ESM">
+
+```javascript {7}
+import { Command } from '@sapphire/framework';
+
+export class PingCommand extends Command {
+  constructor(context) {
+    super(context, {
+      // ...
+      runIn: 'GUILD_ANY' // Only run in guild channels
+    });
+  }
+}
+```
+
+</TabItem>
+<TabItem value="typescript" label="TypeScript">
+
+```typescript {6}
+import { ApplyOptions } from '@sapphire/decorators';
+import { Command, CommandOptions } from '@sapphire/framework';
+
+@ApplyOptions<CommandOptions>({
+  // ...
+  runIn: 'GUILD_ANY' // Only run in guild channels
+})
+export class PingCommand extends Command {}
+```
+
+</TabItem>
+</Tabs>
```

Comment:
This would really benefit from ts2esm2cjs :D

Reply:
She can't do that until #97 is merged and this will likely be merged sooner